File: Capstone/Fracto/Fracto_Frontend/src/app/components/appointment/appointment.ts
import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { AppointmentService } from '../../services/appointment.service';

@Component({
  selector: 'app-appointment',
  standalone: false,
  templateUrl: './appointment.html',
  styleUrls: ['./appointment.css']
})
export class Appointment implements OnInit {
  doctorId!: number;
  appointmentDate: string = '';
  timeSlot: string = '';
  user: any;

  constructor(private route: ActivatedRoute, private appointmentService: AppointmentService) {}

  ngOnInit(): void {
    this.route.queryParams.subscribe(params => {
      this.doctorId = params['doctorId'];
    });
    this.user = JSON.parse(localStorage.getItem('user') || '{}');
  }

  bookAppointment() {
    const appointment = {
      userId: this.user.userId,
      doctorId: this.doctorId,
      appointmentDate: this.appointmentDate,
      timeSlot: this.timeSlot,
      status: 'Pending'
    };

    this.appointmentService.createAppointment(appointment).subscribe(() => {
      alert('Appointment booked successfully!');
    });
  }
}

File: Capstone/Fracto/Fracto_Backend/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models;

public class Appointment
{
    
    public int AppointmentId { get; set; }
    [ForeignKey("User")]
    public int UserId { get; set; }
    public User? User { get; set; }
    [ForeignKey("Doctor")]
    public int DoctorId { get; set; }
    public Doctor? Doctor { get; set; }
    [Required]
    public DateTime AppointmentDate { get; set; }
    [Required]
    public string TimeSlot { get; set; } = string.Empty;
    [Required]
    public string Status { get; set; } = "Pending";
}
